rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -2301,7 +2375,9 @@ static CddsPublisher * create_cdds_publisher(
     set_error_message_from_create_topic(topic, fqtopic_name);
     goto fail_topic;
   }
-  if ((qos = create_readwrite_qos(qos_policies, false)) == nullptr) {
+
+  type_hash = get_type_hash(type_support);
```

Comment:
You could just do:

```suggestion
  rosidl_type_hash_t type_hash = get_type_hash(type_support);
```

Reply:
I did it first this way because it's a compiler error for the declaration to come after the first `goto` in the function. But I'm resolving by just moving the whole thing higher up, in both cases